jvm/core/src/main/scala/org/scalatest/concurrent/AbstractPatienceConfiguration.scala

```diff
@@ -88,10 +89,31 @@ trait AbstractPatienceConfiguration extends ScaledTimeSpans {
    * @author Bill Venners
    * @author Chua Chee Seng
    */
-  final case class PatienceConfig(timeout: Span = scaled(Span(150, Millis)), interval: Span = scaled(Span(15, Millis)))
+  final case class PatienceConfig(timeout: Span = scaled(Span(150, Millis)), interval: Span = scaled(Span(15, Millis))) {
+    /**
+     * <code>PatienceConfig</code> is an inner class. For some reason (like using <code>SocketReporter</code> and having
+     * a <code>ScalaFutures</code> test that fails), instance of this class will be serialized. But as an inner class,
+     * scala will add an extra <code>$outer</code> field that references actual outer class: an instance that you don't
+     * want make serializable.
+     * To avoid errors like <code>java.io.NotSerializableException: org.scalatest.verbs.BehaveWord</code>, this function
+     * delegates serialization to a non inner class.
+     */
+    private def writeReplace(): Any = new PatienceConfigProxy(timeout, interval)
+  }
 
   /**
    * Returns a <code>PatienceConfig</code> value providing default configuration values if implemented and made implicit in subtraits.
    */
   def patienceConfig: PatienceConfig
 }
+
+protected[concurrent] object AbstractPatienceConfiguration extends AbstractPatienceConfiguration {
+  override def patienceConfig: PatienceConfig = PatienceConfig()
```

Comment:
Why is this protected[concurrent] instead of just private[concurrent]?

Reply:
You've right: this visibility can be reduced to `private`. The change is applied.

Reply:
Tnx. Testing this now.